engine/npcyclemanager/update.go

```diff
@@ -27,6 +29,41 @@ func (db *DB) update(ctx context.Context, all bool, alertID *int) error {
 	}
 	log.Debugf(ctx, "Updating notification cycles.")
 
-	_, err = db.lock.Exec(ctx, db.queueMessages)
-	return errors.Wrap(err, "queue outgoing messages")
+	tx, err := db.lock.BeginTx(ctx, nil)
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback()
+
+	rows, err := tx.StmtContext(ctx, db.queueMessages).QueryContext(ctx)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	type record struct {
+		alertID int
+		userID  string
+		meta    *alertlog.NotificationMetaData
+	}
+
+	var data []record
+	for rows.Next() {
+		var rec record
+		err = rows.Scan(&rec.userID, &rec.alertID)
+		rec.meta = &alertlog.NotificationMetaData{}
+		if err != nil {
+			return err
+		}
+		data = append(data, rec)
+	}
+
+	for _, rec := range data {
+		err = db.log.LogTx(ctx, tx, rec.alertID, alertlog.TypeNoNotificationSent, rec.meta)
+		if err != nil {
+			return errors.Wrap(err, "queue outgoing messages")
```

Comment:
Wrap message should be something like "log no notification sent", queuing happens above